### PR TITLE
Job tags

### PR DESCRIPTION
**Pull request by anthony-sarkis**
colors
https://www.loom.com/share/dc1bde3927b94d4684da1b82ab99cb99

SDK
https://github.com/diffgram/python-sdk/pull/37

main backend changes
[shared/database/tag/tag.py](https://github.com/diffgram/diffgram/compare/job-tags?expand=1#diff-5f49fd4df5a223f40f3bea359cbe0248174982d72e58ef13f5a1c12298940da6)

and front end etc

I wish chips were full color like github, but vuetify was being silly so if anyone knows way to make it like github tags that would be better for colors.

### CODE REVIEW

**Review by PJEstrada — COMMENTED** [review]
Thanks Anthony! Looks really good! I'll be addding comments on small thing, all of them can be V2 stuff

File: shared/alembic/versions_opencore/alembic_2022_07_19_16_21_bfb007d3c340_tags.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+# revision identifiers, used by Alembic.
+revision = 'bfb007d3c340'
+down_revision = '625370e1484b'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.create_table('job_tag',
```
Comment (PJEstrada) [inline shared/alembic/versions_opencore/alembic_2022_07_19_16_21_bfb007d3c340_tags.py:20]: MIght be a good idea to add indexes from the getgo to project_id and job id maybe?

File: frontend/src/components/tag/tag_select.vue [hunk 2]
```diff
@@ -0,0 +1,138 @@
+<template>
+  <div v-cloak>
+
+    <v-autocomplete
```
[the remark this answers is not in the document]
Comment (anthony-sarkis) [reply frontend/src/components/tag/tag_select.vue:4]: Thanks Pablo! :)

File: default/methods/task/task_template/job_new_or_update.py [hunk 3]
```diff
@@ -832,6 +848,29 @@ def new_or_update_core(session,
 
 
 
+def add_tags_to_job(tag_list,
+                    session,
+                    project,
+                    job):
+
+
+    if len(tag_list) > 100: return
```
[the remark this answers is not in the document]
Comment (anthony-sarkis) [reply default/methods/task/task_template/job_new_or_update.py]: Good idea! 
added here -> https://github.com/diffgram/diffgram/commit/3c0495df202799e6117a68f01f83e33c383883a7

File: shared/alembic/versions_opencore/alembic_2022_07_19_16_21_bfb007d3c340_tags.py (hunk 1 above)
Comment (anthony-sarkis) [reply shared/alembic/versions_opencore/alembic_2022_07_19_16_21_bfb007d3c340_tags.py:20]: My understanding was that postgres would do that automatically for primary keys https://www.postgresql.org/docs/current/indexes-unique.html
"PostgreSQL automatically creates a unique index when a unique constraint or primary key is defined for a table. The index covers the columns that make up the primary key or unique constraint (a multicolumn index, if appropriate), and is the mechanism that enforces the constraint."

But not 100% sure if that covers your question given the triplicate primary keys

**Pull request merged**